### PR TITLE
Support VNA plan

### PR DESCRIPTION
**Pull request by kajiuray**
This is a commit corresponding to [eclcloud#38](https://github.com/nttcom/eclcloud/pull/38).
The user will be able to create a VNA by indicating some of plan's attribute like `id` and `name`.

This PR includes follows.
- Add `ecl_vna_appliance_plan_v1` data type
- Add a related document
- Add tests

Ref:
https://sdpf.ntt.com/services/docs/firewall/api-references/rsts/APIList/source/list_virtual_network_appliance_plan.html

### CODE REVIEW

**Review by sugimo-s — APPROVED** [review]

File: ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	mc.Register(t, "virtual_network_appliance_plans", "/v1.0/virtual_network_appliance_plans", testMockVNAV1AppliancePlansListNameQuery)
+
+	mc.StartServer(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVNAV1AppliancePlanDataSourceBasic,
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go:26]:
```suggestion
				Config: testAccVNAV1AppliancePlanDataSourceQueryName,
```

File: ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVNAV1AppliancePlanDataSourceID("data.ecl_vna_appliance_plan_v1.appliance_plan_1"),
+					resource.TestCheckResourceAttr(
+						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "name", testAccVNAV1AppliancePlanDataSourcePlanName),
+				),
+			},
+			{
+				Config: testAccVNAV1AppliancePlanDataSourceQueryID,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVNAV1AppliancePlanDataSourceID("data.ecl_vna_appliance_plan_v1.appliance_plan_1"),
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go:64]:
```suggestion
					testAccCheckVNAV1AppliancePlanDataSourceID("data.ecl_vna_appliance_plan_v1.appliance_plan_2"),
```

File: ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+				),
+			},
+			{
+				Config: testAccVNAV1AppliancePlanDataSourceQueryDescription,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVNAV1AppliancePlanDataSourceID("data.ecl_vna_appliance_plan_v1.appliance_plan_1"),
+				),
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go:71]:
```suggestion
					resource.TestCheckResourceAttr(
						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "description", testAccVNAV1AppliancePlanDataSourcePlanName),
				),
```

File: ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        }
+`, testAccVNAV1AppliancePlanDataSourcePlanName,
+)
+
+var testMockVNAV1AppliancePlansListDescriptionQuery = fmt.Sprintf(`
+request:
+    method: GET
+    query:
+      id:
+        - virtual_network_appliance_plans_description
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go:169]:
```suggestion
      description:
        - %q
```

File: ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+response:
+    code: 200
+    body: >
+        {
+          "virtual_network_appliance_plans": [
+            {
+              "id": "37556569-87f2-4699-b5ff-bf38e7cbf8a7",
+              "name": %q,
+              "description": "virtual_network_appliance_plans_description",
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go:178]:
```suggestion
              "description": %q,
```

File: ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+                {
+                  "availability_zone": "zone1_groupb",
+                  "available": false,
+                  "rank": 2
+                }
+              ]
+            }
+          ]
+        }
+`, testAccVNAV1AppliancePlanDataSourcePlanName,
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1_mock_test.go:205]:
```suggestion
`, testAccVNAV1AppliancePlanDataSourcePlanName, testAccVNAV1AppliancePlanDataSourcePlanName, testAccVNAV1AppliancePlanDataSourcePlanName,
```

**Review by ysssasaki — CHANGES_REQUESTED** [review]
I commented on the Mock test.
I will also comment on regular tests later 🙇

File: ecl/data_source_ecl_vna_appliance_plan_v1_test.go [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVNAV1AppliancePlanDataSourceID("data.ecl_vna_appliance_plan_v1.appliance_plan_1"),
+					resource.TestCheckResourceAttr(
+						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "name", testAccVNAV1AppliancePlanDataSourcePlanName),
+					resource.TestCheckResourceAttr(
+						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "description", testAccVNAV1AppliancePlanDataSourcePlanName),
+					resource.TestCheckResourceAttr(
+						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "appliance_type", "ECL::VirtualNetworkAppliance::VSRX"),
+					resource.TestCheckResourceAttr(
+						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "flavor", "2CPU-8GB"),
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1_test.go:29]:
```suggestion
						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "flavor", "2CPU-4GB"),
```

File: ecl/data_source_ecl_vna_appliance_plan_v1_test.go [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+					resource.TestCheckResourceAttr(
+						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "enabled", "true"),
+					resource.TestCheckResourceAttr(
+						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "licenses.#", "1"),
+					resource.TestCheckResourceAttr(
+						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "licenses.0.license_type", "STD"),
+					resource.TestCheckResourceAttr(
+						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "availability_zones.#", "1"),
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1_test.go:37]:
```suggestion
						"data.ecl_vna_appliance_plan_v1.appliance_plan_1", "availability_zones.#", "3"),
```
The value of lab3 is 3.

File: ecl/data_source_ecl_vna_appliance_plan_v1_test.go [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+		return nil
+	}
+}
+
+var testAccVNAV1AppliancePlanDataSourceBasic = fmt.Sprintf(`
+data "ecl_vna_appliance_plan_v1" "appliance_plan_1" {
+  name = %q
+  description = %q
+  appliance_type = "ECL::VirtualNetworkAppliance::VSRX"
+  flavor = "2CPU-8GB"
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1_test.go:91]:
```suggestion
  flavor = "2CPU-4GB"
```

File: ecl/data_source_ecl_vna_appliance_plan_v1_test.go [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+}
+`, testAccVNAV1AppliancePlanDataSourcePlanName, testAccVNAV1AppliancePlanDataSourcePlanName,
+)
+
+var testAccVNAV1AppliancePlanDataSourceQueryID = fmt.Sprintf(`
+data "ecl_vna_appliance_plan_v1" "appliance_plan_1" {
+  name = %q
+}
+data "ecl_vna_appliance_plan_v1" "appliance_plan_2" {
+  id = "${data.ecl_network_appliance_plan_v1.appliance_plan_1.id}"
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1_test.go:110]:
```suggestion
  id = "${data.ecl_vna_appliance_plan_v1.appliance_plan_1.id}"
```

File: ecl/data_source_ecl_vna_appliance_plan_v1.go [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+	d.Set("flavor", plan.Flavor)
+	d.Set("number_of_interfaces", plan.NumberOfInterfaces)
+	d.Set("enabled", plan.Enabled)
+	d.Set("max_number_of_aap", plan.MaxNumberOfAap)
+
+	licenses := make([]map[string]string, len(plan.Licenses))
+	for _, l := range plan.Licenses {
+		license := map[string]string{"license_type": l.LicenseType}
+		licenses = append(licenses, license)
+	}
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1.go:192]:
```suggestion
	licenses := make([]map[string]string, len(plan.Licenses))
	for i, l := range plan.Licenses {
		license := map[string]string{"license_type": l.LicenseType}
		licenses[i] = license
	}
```
The list length is specified at the time of initialization, the length will be doubled when append.

File: ecl/data_source_ecl_vna_appliance_plan_v1.go [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+	d.Set("licenses", licenses)
+
+	availability_zones := make([]map[string]interface{}, len(plan.AvailabilityZones))
+	for _, az := range plan.AvailabilityZones {
+		availability_zone := make(map[string]interface{})
+		availability_zone["availability_zone"] = az.AvailabilityZone
+		availability_zone["available"] = az.Available
+		availability_zone["rank"] = az.Rank
+		availability_zones = append(availability_zones, availability_zone)
+	}
```
Comment (ysssasaki) [inline ecl/data_source_ecl_vna_appliance_plan_v1.go:202]:
```suggestion
	for i, az := range plan.AvailabilityZones {
		availability_zone := make(map[string]interface{})
		availability_zone["availability_zone"] = az.AvailabilityZone
		availability_zone["available"] = az.Available
		availability_zone["rank"] = az.Rank
		availability_zones[i] = availability_zone
	}
```

**Review by ysssasaki — CHANGES_REQUESTED** [review]

**Review by ysssasaki — APPROVED** [review]

**Pull request merged**